libdnf/rpm/solv_query.cpp

```diff
@@ -986,6 +993,357 @@ SolvQuery & SolvQuery::ifilter_location(libdnf::sack::QueryCmp cmp_type, std::ve
+
+    solv::IdQueue out;
+
+    for (auto package_id : *package_set.pImpl) {
+        out.clear();
+
+        // queue_push2 because we are creating a selection, which contains pairs
+        // of <flags, Id>, SOLVER_SOOLVABLE_ALL is a special flag which includes
+        // all packages from specified pool, Id is ignored.
+        queue_push2(&out.get_queue(), SOLVER_SOLVABLE_ALL, 0);
+
+        int flags = 0;
+        flags |= SELECTION_FILTER | SELECTION_WITH_ALL;
+        selection_make_matchsolvable(pool, &out.get_queue(), package_id.id, flags, libsolv_key, 0);
+
+        // Queue from selection_make_matchsolvable is a selection, which means
+        // it conntains pairs <flags, Id>, flags refers to how was the Id
+        // matched, that is not important here, so skip it and iterate just
+        // over the Ids.
+        for (int j = 1; j < out.size(); j += 2) {
+            filter_result.add_unsafe(PackageId(out[j]));
+        }
+    }
+
+    // Apply filter results to query
+    if (cmp_not) {
+        query_result -= filter_result;
+    } else {
+        query_result &= filter_result;
+    }
+}
+
+SolvQuery & SolvQuery::ifilter_conflicts(libdnf::sack::QueryCmp cmp_type, const ReldepList & reldep_list) {
+    p_impl->filter_reldep(SOLVABLE_CONFLICTS, cmp_type, reldep_list);
+    return *this;
+}
+
+SolvQuery & SolvQuery::ifilter_conflicts(libdnf::sack::QueryCmp cmp_type, const std::vector<std::string> & patterns) {
+    p_impl->filter_reldep(SOLVABLE_CONFLICTS, cmp_type, patterns);
+    return *this;
+}
+
+SolvQuery & SolvQuery::ifilter_conflicts(libdnf::sack::QueryCmp cmp_type, const PackageSet & package_set) {
+    p_impl->filter_reldep(SOLVABLE_CONFLICTS, cmp_type, package_set);
+    return *this;
+}
+
+SolvQuery & SolvQuery::ifilter_enhances(libdnf::sack::QueryCmp cmp_type, const ReldepList & reldep_list) {
+    p_impl->filter_reldep(SOLVABLE_ENHANCES, cmp_type, reldep_list);
+    return *this;
+}
+
+SolvQuery & SolvQuery::ifilter_enhances(libdnf::sack::QueryCmp cmp_type, const std::vector<std::string> & patterns) {
+    p_impl->filter_reldep(SOLVABLE_ENHANCES, cmp_type, patterns);
+    return *this;
+}
+
+SolvQuery & SolvQuery::ifilter_enhances(libdnf::sack::QueryCmp cmp_type, const PackageSet & package_set) {
+    p_impl->filter_reldep(SOLVABLE_ENHANCES, cmp_type, package_set);
+    return *this;
+}
+
+SolvQuery & SolvQuery::ifilter_obsoletes(libdnf::sack::QueryCmp cmp_type, const ReldepList & reldep_list) {
+    p_impl->filter_reldep(SOLVABLE_OBSOLETES, cmp_type, reldep_list);
+
+    return *this;
+}
+
+SolvQuery & SolvQuery::ifilter_obsoletes(libdnf::sack::QueryCmp cmp_type, const std::vector<std::string> & patterns) {
+    p_impl->filter_reldep(SOLVABLE_OBSOLETES, cmp_type, patterns);
+    return *this;
+}
+
+SolvQuery & SolvQuery::ifilter_obsoletes(libdnf::sack::QueryCmp cmp_type, const PackageSet & package_set) {
+    bool cmp_not = false;
+    switch (cmp_type) {
+        case libdnf::sack::QueryCmp::EQ:
```

Comment:
Only small speed optimization. Set value `cmp_not = false;` at this place and remove value from declaration above `bool cmp_not;`.

Reply:
Fixed